src/Aspire.Dashboard/Components/Controls/ResourceDetails.razor

```diff
@@ -63,14 +63,50 @@
                         @FilteredEndpoints.Count()
                     </FluentBadge>
                 </div>
-                <PropertyGrid
-                    TItem="DisplayedEndpoint"
-                    Items="@FilteredEndpoints"
-                    ContentAfterValue="(vm) => GetContentAfterValue(vm)"
-                    ValueSort="_endpointValueSort"
-                    IsValueMaskedChanged="@OnValueMaskedChanged"
-                    HighlightText="@_filter"
-                    GridTemplateColumns="1fr 1.5fr" />
+
+                @{
+                    var hasAnyEndpointDisplayNames = FilteredEndpoints.Any(e => e.DisplayName != null);
+                }
+
+                <FluentDataGrid TGridItem="DisplayedEndpoint"
+                                ItemKey="@(vm => ((IPropertyGridItem)vm).Key)"
+                                Items="@FilteredEndpoints"
+                                ColumnResizeLabels="@_resizeLabels"
+                                ColumnSortLabels="@_sortLabels"
+                                HeaderCellAsButtonWithMenu="true"
+                                ResizableColumns="true"
+                                ResizeType="DataGridResizeType.Discrete"
+                                Style="width:100%"
+                                RowSize="DataGridRowSize.Medium"
+                                GridTemplateColumns="@(hasAnyEndpointDisplayNames ? "1fr 1.5fr 1.5fr" : "1.5fr 1.5fr")"
+                                ShowHover="true">
+                    <AspireTemplateColumn Sortable="true" SortBy="@(GridSort<DisplayedEndpoint>.ByAscending(i => i.Name))" Title="@ControlStringsLoc[nameof(ControlsStrings.NameColumnHeader)]">
+                        <GridValue ValueDescription="@ControlStringsLoc[nameof(ControlsStrings.NameColumnHeader)]"
+                                   Value="@context.Name"
+                                   EnableHighlighting="@(!string.IsNullOrEmpty(_filter))"
+                                   HighlightText="@_filter" />
+                    </AspireTemplateColumn>
+                    @if (hasAnyEndpointDisplayNames)
+                    {
+                        <AspireTemplateColumn Sortable="true" SortBy="@(GridSort<DisplayedEndpoint>.ByAscending(i => i.Name))" Title="@ControlStringsLoc[nameof(ControlsStrings.DisplayNameColumnHeader)]">
+                            @if (context.DisplayName is not null)
```

Comment:
Why not have a GridValue if there isn't a display name? The column should display `-` like other empty grid items. I believe GridValue handles doing that for you.

Reply:
Fair point, displaying GridValue now regardless